azure/scope/identity.go

```diff
@@ -102,6 +105,10 @@ func NewAzureClusterCredentialsProvider(ctx context.Context, kubeClient client.C
 
 // GetAuthorizer returns an Azure authorizer based on the provided azure identity. It delegates to AzureCredentialsProvider with AzureCluster metadata.
 func (p *AzureClusterCredentialsProvider) GetAuthorizer(ctx context.Context, resourceManagerEndpoint, activeDirectoryEndpoint, tokenAudience string) (autorest.Authorizer, error) {
+	if err := reconcileASOSecret(ctx, p.AzureCredentialsProvider.Identity, p.Client, p.AzureCluster.Spec.SubscriptionID); err != nil {
```

Comment:
Nope. The intention is to use the credentials ASO controller is configured with and avoid labelling ASO objects that are related to a cluster not specifying the `spec.identityRef`

Reply:
Since subscription ID is specified outside the AzureClusterIdentity, could two clusters to use the same set of default credentials but have different subscription IDs, or is that not possible given how auth generally works in Azure?

Reply:
Yup, it should possible to use the default set of credentials in different subscriptions, as long as the default credentials have access to those subscriptions.

Reply:
Since subscription ID is part of the default ASO credentials, is it possible that we could run into an issue then if an AzureCluster's `spec.identityRef` is undefined and `spec.subscriptionID` is different from ASO's global default? Would we still need an ASO secret in that case to specify the cluster's subscription ID?